packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -163,14 +163,14 @@ export class Rule extends Resource implements IRule {
     // Simply increment id for each `addTarget` call. This is guaranteed to be unique.
     const autoGeneratedId = `Target${this.targets.length}`;
 
-    const targetProps = target.bind(this, autoGeneratedId);
-    const inputProps = targetProps.input && targetProps.input.bind(this);
+    const { input, role, id: propsId, targetResource, arn, ...targetParameters } = target.bind(this, autoGeneratedId);
```

Comment:
@shivlaks I updated this to spread `targetParameters` and pass them below (line 294).

This works, but I am unsure how to make the `RuleTargetConfig` type in `target.ts` be more generic based on `CfnRule` namespaces.

This is still an improvement, I think, because won't have to change source code to add new targets

Reply:
I think it might be better to hold off on these changes / introduce them in a separate PR so we don't mix it up with adding the ApiGateway target. thoughts?